src/scales/scale.time.js

```diff
@@ -334,27 +334,38 @@ function generate(min, max, minor, major, capacity, options) {
  * Returns the right and left offsets from edges in the form of {left, right}.
  * Offsets are added when the `offset` option is true.
  */
-function computeOffsets(table, ticks, min, max, options) {
+function computeOffsets(table, ts, min, max, options) {
 	var left = 0;
 	var right = 0;
-	var upper, lower;
-
-	if (options.offset && ticks.length) {
-		if (!options.time.min) {
-			upper = ticks.length > 1 ? ticks[1] : max;
-			lower = ticks[0];
-			left = (
-				interpolate(table, 'time', upper, 'pos') -
-				interpolate(table, 'time', lower, 'pos')
-			) / 2;
-		}
-		if (!options.time.max) {
-			upper = ticks[ticks.length - 1];
-			lower = ticks.length > 1 ? ticks[ticks.length - 2] : min;
-			right = (
-				interpolate(table, 'time', upper, 'pos') -
-				interpolate(table, 'time', lower, 'pos')
-			) / 2;
+	var timestamps = [];
+	var i, ilen, timestamp, length, upper, lower, divisor;
+
+	if (options.offset) {
+		[ts.labels, ts.data].forEach(function(data) {
+			if (!timestamps.length) {
```

Comment:
Will this work as expected if `ts.data` and `ts.labels` both have items?

It looks what will happen is that the first iteration of the `forEach` will run and `timestamps.length` will be 0 so the items in `ts.labels` are considered. However, the items in `ts.data` will be skipped because on the second iteration, `timestamps.length` may be non-zero due to the first iteration of the loop.

If this is as intended, I think there should be a comment explaining why. I can see it accidentally getting lost/changed later